### PR TITLE
[portal] added toggles for site-types

### LINKED ISSUE #213: Add sidebar, filter by site type

### PR DESCRIPTION
**Pull request by dym-sh**
closes #213, probably

webring-portal--types

### CODE REVIEW

**Comment by neauoire** [comment]
Please follow the coding style of the rest of the project.

**Comment by dym-sh** [comment]
:+1:

**Comment by ckipp01** [comment]
One other thing to think about (I'm also not able to test this at the moment to see how it responds), but it'd be cool if when someone filtered by type, it condensed the list down rather than just hiding the non-matching site types. My main reason for saying this is that now on mobile (I believe), there is going to be huge gaps since all of the sites are going to be in a single column. I think that will be a bad user experience on mobile.

**Comment by dym-sh** [comment]
now uses `display:none` to hide

test under https://dym-sh.github.io/Webring/

webring-portal--types-v2

File: index.html [hunk 1]
```diff
@@ -15,6 +15,15 @@
     const portal = new Portal(sites);
     portal.install(document.body);
     portal.start();
+
+    function toggleType(e) {
```
Comment (ckipp01) [inline index.html]: Are you able to move this out of index.hml and into portal.js? Then, just change the onClick handers to be `portal.toggleType(event)`.

File: scripts/portal.js [hunk 2]
```diff
@@ -16,8 +16,10 @@ function Portal (sites) {
   }
 
   function _directory (sites) {
+    const siteTypesArray = [...new Set(sites.map(site => site.type).filter(Boolean))]
+    const siteTypes = siteTypesArray.reduce((output, siteType) => `${output}<a data-type='${siteType}' href='#' onclick='event.preventDefault(); toggleType(event)'>&lt;${siteType}&gt;</a>`, `<a class='current' href='#' onclick='event.preventDefault(); toggleType(event)'>&lt;all&gt;</a>`)
```
Comment (ckipp01) [inline scripts/portal.js]: I don't believe the `event.preventDefault()` is actually doing anything here is it?

Comment (dym-sh) [reply scripts/portal.js]: it prevents the `#` from href to apply to url

Comment (ckipp01) [reply scripts/portal.js]: Ahhh, duh, of course. Thanks!

File: index.html (hunk 1 above)
Comment (dym-sh) [reply index.html]: sure

Comment (ckipp01) [reply index.html]: 👍

**Comment by ckipp01** [comment]
Looks good, thanks @dym-sh!

**Pull request merged**